src/internal.c

```diff
@@ -5004,7 +5004,9 @@ int EccVerify(WOLFSSL* ssl, const byte* in, word32 inSz, const byte* out,
 #endif /* WOLFSSL_ASYNC_CRYPT */
     {
         if (ret != 0 || ssl->eccVerifyRes == 0) {
-            ret = VERIFY_SIGN_ERROR;
```

Comment:
`if (ret != 0 || ssl->eccVerifyRes == 0)` is false, we don’t need to set ret to 0, it is already. In checks like this I usually check if the verify call succeeds, and then update ret if verifyRes is not-verified. Then check ret for the log message. (This is more of a preference than for any technical reason.)

Reply:
This looks right to me. It will ensure a non-zero ret gets passed back up. However perhaps we should only do this in some build cases. John I agree we need to be careful about this change. Also make sure no downstream code depends on `VERIFY_SIGN_ERROR`.

Reply:
If we can wrap this in the `WOLFSSL_STSAFEA100` define I am okay with it.

Reply:
But we have the callback to isolate the platform specific behavior from wolfSSL.